src/analytics/index.ts

```diff
@@ -1,10 +1,151 @@
-import { createClient } from '@segment/analytics-react-native';
-// @ts-ignore
+import {
+  createClient,
+  JsonMap,
+  SegmentClient,
+} from '@segment/analytics-react-native';
 import { REACT_APP_SEGMENT_API_WRITE_KEY } from 'react-native-dotenv';
+import { TrackingEventProperties, TrackingEvents } from './trackingEvents';
+import { UserProperties } from './userProperties';
+import Routes from '@/navigation/routesNames';
+import { EthereumAddress } from '@/entities';
+import { ethers } from 'ethers';
+import { forEach } from 'lodash';
 
+// TODO: need to generate secure key
+export function secureHmac(value: string) {
+  return ethers.utils.computeHmac(
+    ethers.utils.SupportedAlgorithm.sha256,
+    value,
+    '0x3Cb462CDC5F809aeD0558FBEe151eD5dC3D3f608'
+  );
+}
+
+// TODO: we only use route properties for 1 sheet, we need to collect all possibles and lay out the same as we do for event properties
+// this should live in navigation once we type that
+type RouteKeys = keyof typeof Routes;
+type RouteNames = typeof Routes[RouteKeys];
+
+export class Analytics {
+  private client: SegmentClient;
+  private secureAddressHash: string;
+  public deviceId: string | undefined;
+  public debug: boolean;
+
+  constructor({
+    currentAddress,
+    debug = false,
+  }: {
+    currentAddress: string;
+    debug?: boolean;
+  }) {
+    this.debug = debug;
```

Comment:
Maybe we should use the logger debug functionality here instead 🤔 so to get analytics debug logs, just start your server with `LOG_DEBUG=analytics yarn start:clean`

Reply:
would love to do this, not sure how we would do this without patching the segment package? lmk if u have any ideas